third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -1025,8 +1025,10 @@ impl<'a> FunctionGenerator<'a> {
             if before && ctx.is_alive_before(temp)
                 || !before && ctx.is_alive_after(temp, &[], false)
                 || self.pinned.contains(&temp)
+                || !ctx.fun_ctx.is_droppable(temp)
```

Comment:
Could we just pin all the non-droppable temps as an alternative, given that is common to anywhere in the function (i.e., not specific to a particular program point)? Would be a bit cleaner and abstract.

Reply:
That doesn't seem to be cleaner, since pinned values is something else. For example, they can't be moved. I think this is as clean as it can get, just can't pop non-droppabble from the stack.

Reply:
Ah gotcha - had not considered the non-movable bit about the pinned locals.